actor/actor.go

```diff
@@ -0,0 +1,280 @@
+package actor
+
+import (
+	"context"
+	"sync"
+
+	"github.com/lightningnetwork/lnd/fn/v2"
+)
+
+// ActorConfig holds the configuration parameters for creating a new Actor.
+// It is generic over M (Message type) and R (Response type) to accommodate
+// the actor's specific behavior.
+type ActorConfig[M Message, R any] struct {
+	// ID is the unique identifier for the actor.
+	ID string
+
+	// Behavior defines how the actor responds to messages.
+	Behavior ActorBehavior[M, R]
+
+	// DLO is a reference to the dead letter office for this actor system.
+	// If nil, undeliverable messages during shutdown or due to a full
+	// mailbox (if such logic were added) might be dropped.
+	DLO ActorRef[Message, any]
+
+	// MailboxSize defines the buffer capacity of the actor's mailbox.
+	MailboxSize int
+}
+
+// envelope wraps a message with its associated promise. This allows the sender
+// of an "ask" message to await a response. If the promise is nil, it
+// signifies a "tell" operation (fire-and-forget).
+type envelope[M Message, R any] struct {
+	message M
+	promise Promise[R]
+}
+
+// Actor represents a concrete actor implementation. It encapsulates a behavior,
+// manages its internal state implicitly through that behavior, and processes
+// messages from its mailbox sequentially in its own goroutine.
+type Actor[M Message, R any] struct {
```

Comment:
so everything an actor does is sequential ?

Reply:
Yep, it's fully serial. This handles a lot of the boiler plate that we usually do when making a new "sub-system".